ui/plugin-lifecycle/src/main/services/plugin-manager.service.ts

```diff
@@ -0,0 +1,275 @@
+import { Injectable, Inject } from "@angular/core";
+import { Http, Response } from "@angular/http";
+import { Observable, BehaviorSubject } from "rxjs";
+import { UploadPayload, ChangeScopePlugin } from "../interfaces/Plugin";
+import { PluginValidator } from "../classes/plugin-validator";
+import { ScopeFeedback } from "../classes/ScopeFeedback";
+import { PluginUploaderService } from "./plugin-uploader.service";
+import { PluginPublisher } from "./plugin-publisher.service";
+import { HttpTransferService } from "@vcd/http-transfer-service";
+import { API_ROOT_URL, AuthTokenHolderService } from "@vcd/sdk/common";
+import { UiPluginMetadataResponse, UiPluginMetadata } from "@vcd/bindings/vcloud/rest/openapi/model";
+import { PluginService } from "./plugin.service";
+import { getPropsWithout } from "../helpers/object-helpers";
+import { HttpResponse } from "@angular/common/http";
+import { UiPluginTenantsResponse } from "../interfaces/Tenant";
+
+@Injectable()
+export class PluginManager {
+    private _plugins: UiPluginMetadataResponse[];
+    private _pluginsSubject = new BehaviorSubject<UiPluginMetadataResponse[]>(this._plugins);
+    private _selectedPlugins: UiPluginMetadataResponse[] = [];
+    private _selectedPluginsSubj = new BehaviorSubject<UiPluginMetadataResponse[]>(this.selectedPlugins);
+
+    constructor(
+        @Inject(API_ROOT_URL) private _baseUrl: string = "",
+        private http: Http,
+        private authService: AuthTokenHolderService,
+        private pluginUploaderService: PluginUploaderService,
+        private pluginPublisher: PluginPublisher,
+        private httpTransferService: HttpTransferService,
+        private pluginService: PluginService
+    ) {
+        this.getPluginsList();
+    }
+
+    get baseUrl(): string {
```

Comment:
you shouldn't expose this.  It's already available to other classes via an opaque token

Reply:
Changed in the last commit.